vector.go

```diff
@@ -2,35 +2,11 @@ package chromem
 
 import (
 	"errors"
-	"fmt"
 	"math"
 )
 
 const isNormalizedPrecisionTolerance = 1e-6
 
-// cosineSimilarity calculates the cosine similarity between two vectors.
-// Vectors are normalized first.
-// The resulting value represents the similarity, so a higher value means the
-// vectors are more similar.
-func cosineSimilarity(a, b []float32) (float32, error) {
```

Comment:
I kept it around in case we want to make the pre-normalization during document addition optional, but we can add it back when needed :ok_hand: